Vostok.Hosting/Components/Diagnostics/HealthChecks/DnsResolutionCheck.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Vostok.Hosting.Abstractions.Diagnostics;
+using Vostok.Metrics.System.Process;
+
+namespace Vostok.Hosting.Components.Diagnostics.HealthChecks
+{
+    [PublicAPI]
+    public class DnsResolutionCheck : IHealthCheck, IObserver<CurrentProcessMetrics>
```

Comment:
We don't have a way to dispose of this check and it will leak every time we create it.